src/commands/mod.rs

```diff
@@ -736,6 +774,372 @@ impl DaemonControl {
+            prev_abs_fee
+                .checked_div(prev_tx_size)
+                .ok_or(CommandError::InsaneFees(InsaneFeeInfo::InvalidFeerate))?
+        };
+        // Check replacement transaction's feerate is greater than previous feerate (the only directly conflicting transaction).
+        // TODO: Should we throw error or just set feerate_vb = prev_feerate_vb + 1 and check `rbf_psbt` below?
+        if feerate_vb < prev_feerate_vb.to_sat().checked_add(1).unwrap() {
+            return Err(CommandError::RbfError(RbfErrorInfo::TooLowFeerate(
+                feerate_vb,
+            )));
+        }
+        let prev_txos = &prev_psbt.unsigned_tx.output;
+        let mut destinations: HashMap<bitcoin::Address, bitcoin::Amount> =
+            HashMap::with_capacity(prev_txos.len());
+        let mut change_index: Option<bip32::ChildNumber> = None;
+        for txo in prev_txos {
+            let address = bitcoin::Address::from_script(
+                &txo.script_pubkey,
+                self.config.bitcoin_config.network,
+            )
+            .map_err(CommandError::Address)?;
+            match db_conn.derivation_index_by_address(&address) {
+                Some((ind, true)) => {
+                    // TODO: check if there is more than one change output?
+                    println!("found change at address {} and index {}", address, ind);
+                    change_index = Some(ind);
+                }
+                _ => {
+                    // Non-change outputs are handled in the same way whether they are ours or not.
+                    if !is_cancel {
+                        destinations.insert(address, bitcoin::Amount::from_sat(txo.value));
+                    }
+                }
+            };
+        }
+        // We take all previous inputs as mandatory candidates and, if not self-send, include
+        // confirmed coins as optional.
+        // TODO: For `is_cancel`, shall we include just one previous input?
+        let mut candidate_coins: Vec<CandidateCoin> = prev_coins
+            .into_values()
+            .map(|c| CandidateCoin {
+                coin: c,
+                must_select: true,
+            })
+            .collect();
+        if !destinations.is_empty() {
+            candidate_coins.extend(
+                db_conn
+                    .coins(&[CoinStatus::Confirmed], &[])
+                    .into_values()
+                    .map(|c| CandidateCoin {
+                        coin: c,
+                        must_select: false,
+                    }),
+            );
+        }
+        // Try with increasing feerate until fee paid by replacement transaction is high enough.
+        // Replacement fee must be at least:
+        // sum of fees paid by original transactions + incremental feerate * replacement size.
+        for i in 0..100_u64 {
+            println!("trying feerate of {} with i {}", feerate_vb + i, i);
+            let rbf_psbt = self.create_rbf_spend(
```

Comment:
In case `change_index` is `None`, I can pass an arbitrary value here to avoid incrementing the change index while we try different feerates, and then once the necessary feerate has been found, we can do a final call that increments the change index if necessary.

Reply:
Instead of this initial idea, I instead pass the next change address directly to `create_spend_internal`. It might be that we still increment the change index once in an intermediate call and then the final transaction won't have change, but this approach seems simpler.